src/Util/ClassInfo.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Algolia\SearchBundle\Util;
+
+/**
+ * Retrieves information about a class.
+ *
+ * @internal
+ */
+final class ClassInfo
+{
+    /**
+     * Get class name of the given object.
+     *
+     * @param object $object
+     *
+     * @return string
+     */
+    public static function getClass($object)
+    {
+        return self::getRealClassName(get_class($object));
+    }
+
+    /**
+     * Get the real class name of a class name that could be a proxy.
+     *
+     * @param string $className
+     *
+     * @return string
+     */
+    public static function getRealClassName($className)
+    {
+        // Define variable for static analysis
+        $positionPm = false;
+        // __CG__: Doctrine Common Marker for Proxy (ODM < 2.0 and ORM < 3.0)
```

Comment:
I've seen a test case for the case below in `tests/TestCase/EntityProxyTest.php`, but we're missing a test for the `__CG__` case as far as I can see. Could you add this, please?